internal/storagenode/logstream/committer.go

```diff
@@ -260,40 +259,56 @@ func (cm *committer) commitInternal(cc storage.CommitContext) (err error) {
 	}()
 
 	iter := cm.commitWaitQ.peekIterator()
-	for i := 0; i < numCommits; i++ {
-		llsn := cc.CommittedLLSNBegin + types.LLSN(i)
-		glsn := cc.CommittedGLSNBegin + types.GLSN(i)
+	numLogEntries := 0
+	numCWTs := 0
+	for numLogEntries < numCommits && iter.valid() {
+		cwt := iter.task()
 
-		if uncommttedLLSNBegin+types.LLSN(i) != llsn {
-			err = errors.New("log stream: committer: llsn mismatch")
-			return err
-		}
+		// invariant: numCommits-numLogEntries >= cwt.size
+		for i := range cwt.size {
+			llsn := cc.CommittedLLSNBegin + types.LLSN(numLogEntries)
+			glsn := cc.CommittedGLSNBegin + types.GLSN(numLogEntries)
 
-		// Since the number of tasks in commitWaitQ is inspected above, cwt
-		// must exist.
-		cwt := iter.task()
-		cwt.awg.setGLSN(glsn)
+			if uncommttedLLSNBegin+types.LLSN(numLogEntries) != llsn {
```

Comment:
How about moving to L266?

Reply:
This condition isn't necessary since equality of committed messages' initial LLSN and uncommittedLLSNBegin is already checked. However, this refactoring commit doesn't clean up these for conciseness. I'll clean up `committer.go` soon.